### PR TITLE
add projects table

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-713

#### Links to any Pull Requests related to this

UI - https://github.com/biomage-ltd/ui/pull/245
API - https://github.com/biomage-ltd/api/pull/105
IAC - https://github.com/biomage-ltd/iac/pull/110

#### Anything else the reviewers should know about the changes here

- Created new table `projects` to store `projects`
- Add new global secondary key in `samples` table

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock  (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by StefanBabukov** [comment]
Using the table I see there is a field called 'uuid', which is  a reserved field according to - https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html
image

File: cf/dynamo.yaml [hunk 1]
```diff
@@ -45,7 +45,31 @@ Resources:
       AttributeDefinitions:
         - AttributeName: "experimentId"
           AttributeType: "S"
+        - AttributeName: "projectUuid"
+          AttributeType: "S"
       KeySchema:
         - AttributeName: "experimentId"
           KeyType: "HASH"
+      GlobalSecondaryIndexes:
+        - IndexName: gsiExperimentid
```
Comment (ivababukova) [inline cf/dynamo.yaml]: choose a more descriptive name for this

**Review by ivababukova — APPROVED** [review]

**Comment by aerlaut** [comment]
> Using the table I see there is a field called 'uuid', which is a reserved field according to - https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html
> image

Hmmm... changing this would require quite substantial change in the UI and API. We can go around this by using [Expression Attribute Names](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ExpressionAttributeNames.html), but this might not be good as it's something that everyone has to know.

File: cf/dynamo.yaml (hunk 1 above)
Comment (aerlaut) [reply cf/dynamo.yaml]: changed to `gsiByProjectAndExperimentID`

**Pull request merged**